tests/test__collection_api.py

```diff
@@ -5002,6 +4997,30 @@ def test__aggregate_select_nested(self):
             'select_array': [5, 15],
         }, actual[0])
 
+    def test__aggregate_concatArrays(self):
+        self.db.collection.insert_one({
+            'a': [1, 2],
+            'b': ['foo', 'bar', 'baz'],
+            'c': {
+                'arr1': [123],
+            }
+        })
+        actual = self.db.collection.aggregate([{
+            '$project': {
+                'concat': {'$concatArrays': ['$a', ['#', '*'], '$c.arr1', '$b']},
+                'concat_none': {'$concatArrays': ['$a', None, '$b']},
+                'concat_missing_field': {'$concatArrays': [[1, 2, 3], '$c.arr2']}
```

Comment:
Please add a test trying to concat something that's not an array, especially like `c`. And let's make sure it has the same behaviour than pymongo